core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -2180,7 +2195,7 @@ object Parser {
           // we failed to parse, but didn't consume input
           // is unchanged we continue
           // else we stop
-          errs = errs ++ err
+          errs = errs.map(_ ++ err.value)
```

Comment:
`.value` isn't safe. We should do `for { e1 <- errs; e2 <- err } yield (e1 ++ e2)` or `(errs, err).mapN(_ ++ _)` but maybe the latter is slightly slower due to dispatch via typeclass.